concent_api/core/transfer_operations.py

```diff
@@ -297,14 +297,14 @@ def calculate_token_expiration_deadline(
 ) -> int:
     if operation == FileTransferToken.Operation.upload:
         token_expiration_deadline = (
-            report_computed_task.task_to_compute.compute_task_def['deadline'] +
+            int(report_computed_task.task_to_compute.compute_task_def['deadline']) +
```

Comment:
Shouldn't we round up ? @cameel

Reply:
No, because it's not about rounding. The current situation is that the timestamp can be anything that can be converted to `int`. It can be a string, a float, an int or even a custom class that has a conversion to `int`.

The conversion should be performed using `int()` so this means that in case of floats we truncate not round.

That's not the best design in my opinion. We're talking with Przemek and Łukasz about the possibility of changing it but for the time being that's the rule we must follow.

Reply:
What I have rather meant is that if you do `int(0,9)` you get `0`. So if we use this int value in comparisons, we might allow something to happen before actual deadline in float.

Reply:
No, because the deadline is not a float. If it's anything other than `int`, it's supposed to be converted to `int` before use. So `0.0`, `0.1`, `0.9`, `0.999999` all mean that the deadline is `0`.

It's counter-intuitive and that's why we should stop allowing floats and require it to be an `int` in the message.

Reply:
Ok, just wanted to clarify.

> It's counter-intuitive

True.